### PR TITLE
Cart mb

### PR DESCRIPTION
**Pull request by Maxwell-Baird**
Added a check to make sure the user is login before he can checkout of his cart.
Added a check to make sure the password field is not black when changing you password.

### CODE REVIEW

File: app/views/cart/show.html.erb [hunk 1]
```diff
@@ -46,7 +46,15 @@
 </table>
   <p>Total: <%= number_to_currency(cart.total) %></p>
   <p><%= link_to "Empty Cart", "/cart", method: :delete %></p>
-  <p><%= link_to "Checkout", "/orders/new", method: :get %></p>
+  <% if current_user.role == 0%>
```
Comment (DavidTTran) [inline app/views/cart/show.html.erb:49]: Should we use enum? `current_user.visitor?` looks nicer than `current_user.role == 0`

File: spec/features/merchants/destroy_spec.rb [hunk 2]
```diff
@@ -26,6 +26,9 @@
     end
 
     it "I can't delete a merchant that has orders" do
+      user = User.create(name: "David", address: "123 Test St", city: "Denver", state: "CO", zip: "80204", email: "123@example.com", password: "password", role: 1)
+      allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(user)
```
Comment (DavidTTran) [inline spec/features/merchants/destroy_spec.rb:30]: Nice use of stubs

File: spec/features/profile/edit_spec.rb [hunk 3]
```diff
@@ -49,5 +49,18 @@
...
+
+      visit '/profile'
+      click_link 'Change password'
+
+      fill_in 'password', with:''
+      fill_in 'confirmation', with: ''
+      click_on 'Submit'
+
+
+      expect(page).to have_content("Your password is blank.")
```
Comment (DavidTTran) [inline spec/features/profile/edit_spec.rb:62]: Was this still an error after removing the validate_presence_of :password? Does bcrypt not handle empty passwords?

File: app/views/cart/show.html.erb (hunk 1 above)
Comment (Maxwell-Baird) [reply app/views/cart/show.html.erb:49]: For our next run we can add that in

File: spec/features/profile/edit_spec.rb (hunk 3 above)
Comment (Maxwell-Baird) [reply spec/features/profile/edit_spec.rb:62]: This was left in order to get 100% coverage

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -22,7 +22,10 @@ def update_password
     user = User.find(current_user.id)
     if params[:password] == params[:confirmation]
       user.password = params[:password]
-      if user.save
+      if params[:password] == ''
```
Comment (colintalex) [inline app/controllers/users_controller.rb:25]: Would be to cool to have a module or helper to check for a minimum length, char set, etc. for later!

File: spec/features/profile/edit_spec.rb (hunk 3 above)
Comment (colintalex) [reply spec/features/profile/edit_spec.rb:62]: Glad we did this

**Review by colintalex — COMMENTED** [review]
Looking good!

**Pull request merged**